torch/_inductor/scheduler.py

```diff
@@ -1330,7 +1342,39 @@ def compute_dependencies(self):
         Create dependency edges between nodes, handling aliasing and
         mutation properly.
         """
-        name_to_users: DefaultDict[str, List[NodeUser]] = collections.defaultdict(list)
+
+        T = TypeVar("T")
+
+        class DedupList(Generic[T]):
+            """
+            This data structure behaves like a list except it makes sure the
+            elements remain unique.
+            Normally one could use a set/dict for this purpose however
+            the list in question gets elements appended as it is being
+            iterated over which means that we need to keep the list
+            semantics.
+            """
+
+            def __init__(self, items=None, membership=None):
+                self.items = items or list()
+                self.membership = membership or set()
+
+            def append(self, node_user: T) -> None:
```

Comment:
The dedup logic added to the list is limited to the `append` only. Things like `insert`, `pop`, etc. won't be consistent with the deduping. Do we want to reflect this in the docstring?

Reply:
I did not implement those methods and i am not extending the list implementation here, so these methods will just error